routing/dht/routing.go

```diff
@@ -40,19 +40,24 @@ func (dht *IpfsDHT) PutValue(ctx context.Context, key u.Key, value []byte) error
 		return err
 	}
 
-	peers := dht.routingTable.NearestPeers(kb.ConvertKey(key), KValue)
-
-	query := newQuery(key, dht.network, func(ctx context.Context, p peer.ID) (*dhtQueryResult, error) {
```

Comment:
@whyrusleeping did you find problems in the query or something? or why is it removed in favor of a waitgroup approach? if there's a problem in the query runner let's find it, side-stepping pushes the problem elsewhere. Also do note that `PutValue` is now incorrect, as `PutValue` needs to make sure to put the value to all the closest peers it can find (a query), not just the ones it has in the routing table currently. (it's these queries on get/put that make sure the routing table is actually good. in the avg case the queries will be short because we'll have a good routing table)